packages/worker/src/worker-options.ts

```diff
@@ -411,13 +416,100 @@ export function addDefaultWorkerOptions(options: WorkerOptions): WorkerOptionsWi
     // 4294967295ms is the maximum allowed time
     isolateExecutionTimeout: debugMode ? '4294967295ms' : '5s',
     workflowThreadPoolSize: 8,
-    maxCachedWorkflows: maxCachedWorkflows ?? Math.max(os.totalmem() / GiB - 1, 1) * 200,
+    maxCachedWorkflows:
+      maxCachedWorkflows ??
+      Math.floor(Math.max((systemResources.memory + systemResources.swap - systemResources.heap) / GiB - 1, 1) * 250),
```

Comment:
For each new workflow in the cache, is the memory taken mostly in or out of heap? I think @Sushisource said there's not much incremental memory usage in Rust? Our apidoc says the largest memory footprint is the code bundle size.

^ determines what the numerator should be, and next question is how much should be subtracted (currently 1GB) for non-workflow-cache (keeping in mind if we have a 50% heap recommendation, then much of the rest is also non-workflow-cache)

Reply:
The incremental memory usage in Rust is very small. Something on the order of a few kB at most. So it can effectively be ignored for the purposes of this calculation I think. There might be a spot or two where having large payloads can change that but, that's not really predictable.

On another note - if `swap` here means what it normally means (not sure exactly what it means in the container context based on the other code), then we might actually want to ignore it for the purposes of this calculation. If we allow caching until normal memory is filled and we start using swap, things are just gonna be super slow

Reply:
Yeah... I'll remove that. I was working under the incorrect assumption that once a workflow completed its activation, it associated objects in the nodejs heap were immediately disposed, and that the rust core only kept history in the mean time. I read the code more in depth this morning to finally understand that it is core that nodejs resources for a workflow only get disposed once the workflow is removed from `RunCache`.

Reply:
Probably don't need to consider the heap size limit here, just use the memory available, most of the resources are going to be consumed by the node heap.